Squares.Domain/Entities/SquaresDetectionEntry.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Squares.Domain.Entities
+{
+    public class SquaresDetectionEntry
+    {
+        public string Id { get; set; }
+        public IList<Point> InputPoints { get; set; }
```

Comment:
It would be a good idea if the `InputPoints` & `DetectedSquares` properties were initialized to an empty list to avoid null reference exceptions.

Then in `line 25` it is not necessary anymore to initialize `DetectedSquares` property.